lib/elixir/src/elixir_parser.yrl

```diff
@@ -701,8 +701,16 @@ build_fn(Op, _Stab) ->
 
 build_access(Expr, {List, Location}) ->
   Meta = meta_from_location(Location),
+  ok = assert_access_available(Meta, Expr),
   {{'.', Meta, ['Elixir.Access', get]}, Meta, [Expr, List]}.
 
+assert_access_available(_Meta, nil) -> ok;
+assert_access_available(Meta, Expr)
+when is_binary(Expr); is_number(Expr); is_atom(Expr);
+     element(1, Expr) == '{}'; element(1, Expr) == '<<>>' ->
+  throw(Meta, "the Access is not available for the literal before: ", "'['");
```

Comment:
The token should not be `"'['"`. I think we need to raise a regular error and not a parser error. Another option is to not change here and change in elixir_rewrite to raise on `Access.get`.